session_rpcserver.go

```diff
@@ -979,16 +998,57 @@ func (s *sessionRpcServer) AddAutopilotSession(ctx context.Context,
 		caveats = append(caveats, firewall.MetaPrivacyCaveat)
 	}
 
+	// If a previous session ID has been set to link this new one to, we
+	// first check if we have the referenced session, and we make sure it
+	// has been revoked.
+	var prevSess *session.Session
+	if len(req.PrevSessId) != 0 {
+		var prevSessID session.ID
+		copy(prevSessID[:], req.PrevSessId)
+
+		// Make sure that the previous session does actually exist.
+		prevSess, err = s.cfg.db.GetSessionByID(prevSessID)
+		if errors.Is(err, session.ErrSessionNotFound) {
+			return nil, fmt.Errorf("linked session(%x) not found",
+				req.PrevSessId)
+		} else if err != nil {
+			return nil, err
+		}
+
+		// Now we need to check that the linked session is no longer
+		// active.
+		if prevSess.State != session.StateRevoked &&
+			prevSess.State != session.StateExpired {
+
+			return nil, fmt.Errorf("linked session(%x) "+
```

Comment:
>  Perhaps we can allow revoking of sessions through the session_id in a later PR as well :

Yes was gonna add this to this PR but think it is better to just do it in a follow up to not expand the scope too much. Think we should provide a way to revoke by both session_ID and group ID

Reply:
> (Image if the user has 100s of sessions linked)

Yeah but it will be the latest session in the group. So not too hard to grep for at the moment. Can add helpers for this later

Reply:
> Think we should provide a way to revoke by both session_ID and group ID

Agree, let's leave this for a follow-up :)!